core/src/main/java/hudson/Util.java

```diff
@@ -1187,7 +1187,21 @@ public static void copyFile(@Nonnull File src, @Nonnull File dst) throws BuildEx
         cp.setTofile(dst);
         cp.setFile(src);
         cp.setOverwrite(true);
+        cp.setPreserveLastModified(true);
         cp.execute();
+        try {
+            copyFilePosixPermissions(src, dst);
+        } catch (IOException e) {
+            throw new BuildException("Cannot copy file permissions from " + src + " to " + dst, e);
+        }
+    }
+
+    /**
+     * Copies the posix file permissions from one file into another file.
+     */
+    public static void copyFilePosixPermissions(@Nonnull File src, @Nonnull File dst) throws IOException {
```

Comment:
For new public APIs, you want to add `@Restricted(NoExternalUse.class)` so that plugins using the standard parent POM are not able to use the new API. If the whole point of the public API is for plugins to use it then add `@since TODO` to the JavaDoc and someone will update it with the version it was released in after it is released.

In this case I am somewhat included to not add the function at all and instead prefer using something like `Files.copy(fileToPath(src), fileToPath(dest), StandardCopyOption.COPY_ATTRIBUTES)` directly at call sites.

Reply:
Using `COPY_ATTRIBUTES` certainly looks simpler! It may even take care of the Windows-specific attributes. I'll investigate.